src/commands/createNewProject.ts

```diff
@@ -100,45 +147,137 @@ const localSettingsJson: {} = {
+    const version: string = await ui.showInputBox(versionPlaceHolder, versionPrompt, false, undefined, '1.0-SNAPSHOT');
+
+    const packagePlaceHolder: string = localize('azFunc.java.packagePlaceHolder', 'Package');
+    const packagePrompt: string = localize('azFunc.java.packagePrompt', 'Provide value for package');
+    const packageName: string = await ui.showInputBox(packagePlaceHolder, packagePrompt, false, undefined, groupId);
+
+    const appNamePlaceHolder: string = localize('azFunc.java.appNamePlaceHolder', 'App Name');
+    const appNamePrompt: string = localize('azFunc.java.appNamePrompt', 'Provide value for appName');
+    const appName: string = await ui.showInputBox(appNamePlaceHolder, appNamePrompt, false, undefined, `${artifactId}-${Date.now()}`);
+
+    return {
+        groupId: groupId,
+        artifactId: artifactId,
+        version: version,
+        packageName: packageName,
+        appName: appName
+    };
+}
+
 export async function createNewProject(outputChannel: OutputChannel, functionAppPath?: string, openFolder: boolean = true, ui: IUserInterface = new VSCodeUI()): Promise<void> {
     if (functionAppPath === undefined) {
         functionAppPath = await workspaceUtil.selectWorkspaceFolder(ui, localize('azFunc.selectFunctionAppFolderNew', 'Select the folder that will contain your function app'));
     }
 
+    const languages: Pick[] = [
+        new Pick(TemplateLanguage.JavaScript),
+        new Pick(TemplateLanguage.Java)
+    ];
+    const language: string = (await ui.showQuickPick(languages, localize('azFunc.selectFuncTemplate', 'Select a language for your function project'))).label;
+
+    let javaTargetPath: string = '';
+    switch (language) {
+        case TemplateLanguage.Java:
+            // Get parameters for Maven command
+            const { groupId, artifactId, version, packageName, appName } = await promotForMavenParameters(ui, functionAppPath);
+            // Use maven command to init Java function project.
+            await executeCommand(
+                outputChannel,
+                functionAppPath,
+                'mvn',
+                'archetype:generate',
+                '-DarchetypeGroupId="com.microsoft.azure"',
+                '-DarchetypeArtifactId="azure-functions-archetype"',
+                `-DgroupId="${groupId}"`,
+                `-DartifactId="${artifactId}"`,
+                `-Dversion="${version}"`,
+                `-Dpackage="${packageName}"`,
+                `-DappName="${appName}"`,
+                '-B' // in Batch Mode
+            );
+
+            functionAppPath = path.join(functionAppPath, artifactId);
+            javaTargetPath = `target/azure-functions/${appName}/`;
+            break;
+        default:
+            // the maven archetype contains these files, so not check them when language is Java
+            const hostJsonPath: string = path.join(functionAppPath, 'host.json');
```

Comment:
Why does the maven archetype have these files? Shouldn't it let VS Code (or the func cli) handle it? Same with the `.gitignore`

Reply:
Do you own the maven archetype or does someone else? I thought you guys created it